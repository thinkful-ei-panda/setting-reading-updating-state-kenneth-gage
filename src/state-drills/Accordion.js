import React from "react";
import './Accordion.css';

class Accordion extends React.Component {

	static defaultProps = { sections: [] };

	state = { currentSectionIndex: 0 };

	handleButtonClick = ( index ) => {

		this.setState ({ currentSectionIndex: index })

	}

	renderButtons ( section, index ) {

		return (

			<button key={ index } onClick={ () => this.handleButtonClick ( index ) }>

					{ section.title }

			</button>

		)

	}

	renderContent ( section, index ) {

		return ( ( this.state.currentSectionIndex === index ) && <p>{section.content}</p> )
		
	}

	renderLi () {

		return this.props.sections.map ( ( section, index ) => (

			<li className="accordion-item" key={ index }>

				{ this.renderButtons ( section, index ) }

				{ this.renderContent ( section, index ) }

			</li>

		))

	}
	
	render () {

		return (

			<div>

				<ul>

					{ this.renderLi () }
					
				</ul>

			</div>

		)

	}

}

export default Accordion;