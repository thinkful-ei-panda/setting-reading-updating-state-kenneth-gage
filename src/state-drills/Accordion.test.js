import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

describe ( `Accordion Component`, () => {
	// array of objects, each with a name and content
	const sectionsProp = [
		{
		  title: 'Section 1',
		  content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
		},
		{
		  title: 'Section 2',
		  content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
		},
		{
		  title: 'Section 3',
		  content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
		},
	  ]

	it ( 'renders without errors', () => {

		const div = document.createElement ( 'div' );
	
		ReactDOM.render ( <Accordion />, div );
	
		ReactDOM.unmountComponentAtNode ( div );
	
  	})

	it ( 'renders empty given no tabs', () => {

		const wrapper = shallow ( <Accordion /> )
		
		expect ( toJson ( wrapper ) ).toMatchSnapshot ()

	})

	it ( 'renders the first tab by default', () => {

		const wrapper = shallow ( <Accordion sections = { sectionsProp } /> )
		
		expect ( toJson ( wrapper ) ).toMatchSnapshot ()

	})

	it ( 'closes the first tab and opens any clicked tab', () => {
		
		const wrapper = shallow ( <Accordion sections = { sectionsProp } /> )

		wrapper.find ( 'button' ).at ( 1 ).simulate ( 'click' )

		expect ( toJson ( wrapper ) ).toMatchSnapshot ()

	  })

})