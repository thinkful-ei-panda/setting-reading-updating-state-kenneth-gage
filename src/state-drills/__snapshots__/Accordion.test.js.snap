// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component closes the first tab and opens any clicked tab 1`] = `
<div>
  <ul>
    <li
      className="accordion-item"
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      className="accordion-item"
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      className="accordion-item"
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion Component renders empty given no tabs 1`] = `
<div>
  <ul />
</div>
`;

exports[`Accordion Component renders the first tab by default 1`] = `
<div>
  <ul>
    <li
      className="accordion-item"
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      className="accordion-item"
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      className="accordion-item"
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;
